src/app/contacts.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { CONTACTS, Contact } from './contacts/contacts';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContactsService {
+  contacts: Contact[] = CONTACTS;
+
+  getContacts(): Contact[] | undefined {
+    // Get the contacts array
+    // Input: n/a
+    // Return: List of Contact or undefined
+    return this.contacts;
+  }
+  getContact(contactId: number): Contact | undefined {
+    // Get the object with provided id from contacts array
+    // Input: contactId (number)
+    // Return: Contact or undefined
+    return this.contacts.find((contact) => contact.id === contactId);
+  }
+  createContact(
+    firstName: string,
+    lastName: string,
+    street: string,
+    city: string
+  ) {
+    // Create an object with provided data and push into contacts array
+    // Input: firstName (string), phoneNumber (string)
+    // Return: n/a
+    const newContact: Contact = {
+      id: this.contacts.length + 1,
```

Comment:
While this works in many cases, it might not always guarantee a unique ID, especially if contacts are ever deleted from the array. Consider implementing a more robust way to generate unique IDs, perhaps using a timestamp or a separate counter that increments independently of the array's current length.